### PR TITLE
ROS2 port of feature to report stale when no errors but stale items

### PR DESCRIPTION
**Pull request by MCFurry**
ROS2 port of: https://github.com/nobleo/diagnostics/pull/2/files

### CODE REVIEW

**Comment by MCFurry** [comment]
Not sure why rolling is failing, compiles fine locally in a rolling docker...

**Comment by Timple** [comment]
Isn't this the same as: https://github.com/ros/diagnostics/pull/315
Which is already merged and released upstream?

**Comment by MCFurry** [comment]
Not sure why rolling is failing, compiles fine locally in a rolling docker...

> Isn't this the same as: [ros#315](https://github.com/ros/diagnostics/pull/315) Which is already merged and released upstream?

Nope, that PR specifically targets discard analyzers. This PR makes sure that a stale diagnostic makes the toplevel output stale instead of error.

**Comment by Timple** [comment]
> Not sure why rolling is failing, compiles fine locally in a rolling docker...

Can rebase the `nobleo-ros2` branch to the `ros2` branch such that we have a clean start again? Some of our PR's did get merged like the cpu monitor and stuff.

That might fix the CI as well would be my guess.

**Comment by MCFurry** [comment]
> > Not sure why rolling is failing, compiles fine locally in a rolling docker...
> 
> Can rebase the `nobleo-ros2` branch to the `ros2` branch such that we have a clean start again? Some of our PR's did get merged like the cpu monitor and stuff.
> 
> That might fix the CI as well would be my guess.

That would be a good idea indeed!

**Comment by MCFurry** [comment]
Syncing done, let's see what the CI says now

**Review by Timple — APPROVED** [review]

File: diagnostic_aggregator/src/aggregator.cpp [hunk 1]
```diff
@@ -256,6 +259,7 @@ void Aggregator::publishData()
     diag_toplevel_state.values = msg_to_report->values;
   }
 
+  non_ok_status_depth = 0;
```
Comment (Rayman) [inline diagnostic_aggregator/src/aggregator.cpp:262]: What is this new variable?

**Review by Rayman — APPROVED** [review]

File: diagnostic_aggregator/src/aggregator.cpp [hunk 2]
```diff
@@ -221,7 +221,7 @@ void Aggregator::publishData()
   diag_toplevel_state.name = "toplevel_state";
   diag_toplevel_state.level = DiagnosticStatus::STALE;
   int max_level = -1;
-  int min_level = 255;
+  int8_t max_level_without_stale = 0;
   int non_ok_status_depth = 0;
```
Comment (MCFurry) [inline diagnostic_aggregator/src/aggregator.cpp:225]: @Rayman it is here already and simply reset on line 262.

**Pull request merged**